crowd_anki/importer/anki_importer.py

```diff
@@ -62,21 +54,59 @@ def load_from_directory(self, directory_path, import_media=True):
     def get_deck_path(self, directory_path):
         """
         Provides compatibility layer between deck file naming conventions.
+        Assumes that deck json file is located in the directory and named 'deck.json'
         """
 
         def path_for_name(name):
             return directory_path.joinpath(name).with_suffix(DECK_FILE_EXTENSION)
 
-        convention_path = path_for_name(self.deck_file_name)
-        inferred_path = path_for_name(directory_path.name)
+        convention_path = path_for_name(self.deck_file_name)   # [folder]/deck.json
+        inferred_path = path_for_name(directory_path.name)     # [folder]/[folder].json
         return convention_path if convention_path.exists() else inferred_path
 
+    def load_deck_with_settings(self, directory_path) -> (dict, ImportConfig):
+        deck_json = self.read_deck(self.get_deck_path(directory_path))
+        import_config = self.read_import_config(directory_path)
+
+        import_dialog = ImportDialog(deck_json, import_config)
+        if import_dialog.exec_() == QDialog.Rejected:
+            return None, None  # User has cancelled
+
+        # TODO: strip settings from deck_json
+
+        return deck_json, import_dialog.final_import_config
+
     @staticmethod
-    def import_deck_from_path(collection, directory_path, import_media=True):
+    def read_deck(file_path):
+        """
+        Load deck from json file
+        :type file_path: Path
+        """
+        if not file_path.exists():
+            raise ValueError("There is no {} file inside of the selected directory".format(file_path))
+
+        with file_path.open(encoding='utf8') as deck_file:
+            return json.load(deck_file)
+
+    @staticmethod
+    def read_import_config(directory_path):
+        file_path = directory_path.joinpath(IMPORT_CONFIG_NAME).with_suffix(CONFIG_EXTENSION)
+
+        if not file_path.exists():
+            return {}
+
+        with file_path.open(encoding='utf8') as meta_file:
+            return yaml.full_load(meta_file)
+
+    @staticmethod
+    def import_deck_from_path(collection, directory_path):
         importer = AnkiJsonImporter(collection)
         try:
-            importer.load_from_directory(directory_path, import_media)
-            aqt.utils.showInfo("Import of {} deck was successful".format(directory_path.name))
+            deck_json, import_config = importer.load_deck_with_settings(directory_path)
+
+            if deck_json is not None:
+                importer.load_deck(deck_json, directory_path, import_config=import_config)
+                aqt.utils.showInfo("Import of {} deck was successful".format(directory_path.name))
```

Comment:
the way this is structured right now does not make sense to me. I think it should be something like:  
* have a  top level function like `load_deck` that gets the directory path
* internally it loads config (function) and gets update from the user (function)
* then it loads json and saves it to the collection (function)
* then it loads media if required (another function)

Reply:
It was structured this was as the import_config was inside the json file, so I had to read that first. I believe I can restructure it as stated :+1:

Reply:
Oh no, nevermind. I need to read in the deck in order to populate the note models as Personal Field options.